Eshop.WebApi/Controllers/CategoryController.cs

```diff
@@ -16,38 +24,46 @@ public class CategoryController : Controller
         [HttpGet]
         public List<Category> GetCategories()
         {
-            return Categories;
+            dbContext.SaveChanges();
+
+            return dbContext.Categories.ToList(); ;
         }
 
         [HttpGet("{id}")]
         public Category GetCategory(int id) 
-        { 
-            return Categories.First(x => x.Id == id);
+        {
+            dbContext.SaveChanges();
+
+            return dbContext.Categories.First(x => x.Id == id);
         }
 
         [HttpPost]
         public Category CreateCategory(string name) 
         { 
-            var newId = Categories.Max(x => x.Id) + 1;
-            var newCategory = new Category(newId, name);
+            var newCategory = new Category(0, name);
 
-            Categories.Add(newCategory);
+            dbContext.Categories.Add(newCategory);
+            dbContext.SaveChanges();
 
             return newCategory;
         }
 
         [HttpDelete("{id}")]
         public void DeleteCategory(int id) 
         { 
-            var categoryToBeDeleted = Categories.First(x => x.Id == id);
-            Categories.Remove(categoryToBeDeleted);
+            var categoryToBeDeleted = dbContext.Categories.First(x => x.Id == id);
+            dbContext.Categories.Remove(categoryToBeDeleted);
+
+            dbContext.SaveChanges();
         }
 
         [HttpPut("{id}")]
         public Category UpdateCategory(int id, string Name) 
```

Comment:
parameter must be camelCase, Name-> name

Reply:
fixed